nkms/config/keys.py

```diff
@@ -253,18 +263,21 @@ def generate(cls, passphrase: str, encryption: bool=True, transacting: bool=True
         # Create the key directories with default paths. Raises OSError if dirs exist
         os.mkdir(cls.__default_keyring_root, mode=0o755)    # keyring
         os.mkdir(cls.__default_public_key_dir, mode=0o744)  # public
-        os.mkdir(_private_key_dir, mode=0o744)              # private
+        os.mkdir(_private_key_dir, mode=0o700)              # private
 
         # Generate keys
         keyring_args = dict()
         if encryption is True:
-            enc_key, _ = _generate_encryption_keys()
-            sig_key, _ = _generate_signing_keys()
+            enc_privkey, enc_pubkey = _generate_encryption_keys()
```

Comment:
Let's move this logic into a private method and call it here.

Reply:
:-1: Until this gets cleaned up more, I don't really feel like moving the code to another function will help things. :/